bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/widgets/WidgetWin32Tests.java

```diff
@@ -28,8 +36,280 @@ public void testWidgetZoomShouldChangeOnZoomLevelChange() {
 		button.setText("Widget Test");
 		button.setBackground(shell.getDisplay().getSystemColor(SWT.COLOR_CYAN));
 		shell.open();
-		assertEquals("The initial zoom is wrong", zoom, button.getZoom()); //pre-condition
+		assertEquals("The initial zoom is wrong", zoom, button.getZoom()); // pre-condition
+		changeDPIZoom(scaledZoom);
+		assertEquals("The Zoom Level should be updated for button on zoom change event on its shell", scaledZoom,
+				button.getZoom());
+	}
+
+	@Test
+	public void testButtonPointsAfterZooming() throws NoSuchMethodException, IllegalAccessException {
+		int zoom = DPIUtil.getDeviceZoom();
+		int scaledZoom = zoom * 2;
+
+		shell = new Shell(display);
+		shell.setBounds(0, 0, 100, 160);
+		shell.setLayout(new FillLayout());
+		shell.pack();
+
+		Button button = new Button(shell, SWT.PUSH);
+		button.setText("Button");
+		button.setBounds(0, 0, 100, 200);
+		Point sizeBeforeEvent = button.getSize();
+		Point p1 = button.computeSizeInPixels(sizeBeforeEvent.x, sizeBeforeEvent.y, false);
 		changeDPIZoom(scaledZoom);
-		assertEquals("The Zoom Level should be updated for button on zoom change event on its shell", scaledZoom, button.getZoom());
+		Point sizeAfterEvent = button.getSize();
+		Point p2 = button.computeSizeInPixels(sizeAfterEvent.x, sizeAfterEvent.y, false);
+
+		assertEquals("Width should be half in points after zooming to 200", p1.x, p2.x * 2);
+		assertEquals("Height should be half in points after zooming to 200", p1.y, p2.y * 2);
```

Comment:
Do not alter the **actual** value, it's a bad practice.

```suggestion
		assertEquals("Width should be half in points after zooming to 200", p1.x / 2, p2.x);
		assertEquals("Height should be half in points after zooming to 200", p1.y / 2, p2.y);
```